Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -338,10 +347,51 @@ namespace PhysX
     }
 
     // DisplayCallback
-    void EditorHeightfieldColliderComponent::Display(AzFramework::DebugDisplayRequests& debugDisplay) const
+    void EditorHeightfieldColliderComponent::Display(const AzFramework::ViewportInfo& viewportInfo,
+        AzFramework::DebugDisplayRequests& debugDisplay) const
     {
-        const auto& heightfieldConfig = static_cast<const Physics::HeightfieldShapeConfiguration&>(*m_shapeConfig);
-        m_colliderDebugDraw.DrawHeightfield(debugDisplay, m_colliderConfig, heightfieldConfig);
+        AzPhysics::SimulatedBody* simulatedBody = GetSimulatedBody();
+        if (!simulatedBody)
+        {
+            return;
+        }
+
+        AzPhysics::StaticRigidBody* staticRigidBody = azrtti_cast<AzPhysics::StaticRigidBody*>(simulatedBody);
+        if (staticRigidBody->GetShapeCount() == 1)
```

Comment:
Is there any chance that there are more than 1 shape? Maybe the comparison would be more future proof if it were >=1?

Reply:
It'd be pretty weird to have more than 1 heightfield shape at the moment since they'd overlap with each other, but you're right, this code doesn't have to rely on it being just 1 shape. It's possible we'll add the offsets support to heightfields and will be able to have multiple. 
I'll make it a for-loop instead, no other change is needed, thanks!

Reply:
Nice change.